src/nvd/task/check.clj

```diff
@@ -65,10 +65,16 @@
           (throw exception-info))))
     project))
 
-(defn conditional-exit [project]
-  (if (:exit-after-check project)
-    (System/exit (if (:failed? project) -1 0))
-    project))
+(defn conditional-exit [{:keys [throw-if-check-unsuccessful? exit-after-check failed?]
+                         :as project}]
+  (cond
+    throw-if-check-unsuccessful?
```

Comment:
should `failed?` be incorporated into this condition otherwise it would always throw?